### PR TITLE
NXP-21289 Add Blob encoding in Content-Type header

### PR DESCRIPTION
**Pull request by egiuly-nuxeo**
https://jira.nuxeo.com/browse/NXP-21289

### CODE REVIEW

File: nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/ResponseHelper.java [hunk 1]
```diff
@@ -79,11 +80,18 @@ public static Response blob(Blob blob, int httpStatus) {
...
+        Response response = Response.status(httpStatus)
                        .entity(blob)
                        .type(type)
                        .header("Content-Disposition", "attachment; filename=" + blob.getFilename())
                        .build();
+        // set Content-Type header with mimetype and charset
+        if (blob.getEncoding() != null) {
+            List<Object> contentType = new ArrayList<Object>();
+            contentType.add(type + "; charset=" + blob.getEncoding());
+            response.getMetadata().put("Content-Type", contentType);
```
Comment (kevinleturc) [inline nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/ResponseHelper.java]: Use here Collections.singletonList("...")

Comment (efge) [inline nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/ResponseHelper.java]: No need to change the response after it's built. In the original code just do:

    if (blob.getEncoding() != null) {
        type += "; charset=" + blob.getEncoding();
    }

**Comment by egiuly-nuxeo** [comment]
Thanks @kevinleturc and @efge for your feedback

**Comment by efge** [comment]
👍

**Pull request closed without merging**